allauth/socialaccount/providers/base/views.py

```diff
@@ -1,11 +1,13 @@
+from typing import Optional
+
 from django.views import View
 
 from allauth.socialaccount.adapter import get_adapter
 from allauth.socialaccount.providers.base.utils import respond_to_login_on_get
 
 
 class BaseLoginView(View):
-    provider_id = None
+    provider_id: Optional[str] = None
```

Comment:
The call to `.get_provider()` down always requires a `provider_id` -- typing wise this should not be optional.

Reply:
Sure. Looks like `saml.views.LoginView` doesn't set `provider_id` though?

Reply:
Correct -- but it overrides the `get_provider()` method. If you make `provider_id` optional, I would expect `mypy` to complain about this method:

```
    def get_provider(self):
        provider = get_adapter().get_provider(self.request, self.provider_id)
        return provider
```

... as here we're passing `self.provider_id` to a method that does not allow `None`. SAML overrides that method.